pandas/core/frame.py

```diff
@@ -7015,10 +7015,13 @@ def append(self, other, ignore_index=False, verify_integrity=False, sort=None):
             other = other._convert(datetime=True, timedelta=True)
             if not self.columns.equals(combined_columns):
                 self = self.reindex(columns=combined_columns)
-        elif isinstance(other, list) and not isinstance(other[0], DataFrame):
-            other = DataFrame(other)
-            if (self.columns.get_indexer(other.columns) >= 0).all():
-                other = other.reindex(columns=self.columns)
+        elif isinstance(other, list):
```

Comment:
this actually should be is_list_like

Reply:
I took a look at is_list_like - it seems to accept any `abc.Iterable` except `str, bytes`, `pd.Series` and Numpy arrays. The docs for the append method don't mention Numpy arrays nor tuples as valid inputs `other : DataFrame or Series/dict-like object, or list of these`.

Any tuple currently getting passed to the append method would fail at this part https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L7017-L7018 with a `TypeError: can only concatenate list (not "tuple") to list`.

I'm not entirely sure why that condition is checking for a tuple as it will always fail. It might make more sense to remove it.

@jreback thoughts? and whether it still makes sense to make a test that parametrizes across empty tuple, list and array

Reply:
actually try to replace this with is_list_like and see if anything fails;

Reply:
replacing it with `is_list_like` causes test cases to fail in the next line of my current change `if not other` when a DataFrame gets passed in.

```py
        elif is_list_like(other):
            if not other:
                pass
            elif not isinstance(other[0], DataFrame):
                other = DataFrame(other)
                if (self.columns.get_indexer(other.columns) >= 0).all():
                    other = other.reindex(columns=self.columns)
```

I'm not sure of a check that would return False in this case (where `other` is empty) without resorting to another `isinstance` check. Essentially this block is handling the case where `other` should be a list of Series/dict